cabal-install/src/Distribution/Client/CmdUpdate.hs

```diff
@@ -197,10 +197,12 @@ updateRepo verbosity _updateFlags repoCtxt (repo, indexState) = do
       -- NB: always update the timestamp, even if we didn't actually
       -- download anything
       writeIndexTimestamp index indexState
-      ce <- if repoContextIgnoreExpiry repoCtxt
-              then Just `fmap` getCurrentTime
-              else return Nothing
-      updated <- Sec.uncheckClientErrors $ Sec.checkForUpdates repoSecure ce
+      -- typically we get the current time to check expiry against
+      -- but if the flag is set, we don't.
+      now <- case repoContextIgnoreExpiry repoCtxt of
```

Comment:
This actually causes name-shadowing just like in the 3.10 commit: https://github.com/haskell/cabal/commit/dcfdc9cffd74cade4e8cf3df37c5993413ffd30f

See line 213.

Reply:
Sorry. I did this late at night. I'll push a fix but I won't be able to do it today.
I won't be offended if someone else did it before me ☺️